File: SilverlightApplication1/SilverlightApplication1/Observables/ObserverCollection.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Collections.ObjectModel;

namespace SilverlightApplication1
{
    public class ObserverCollection<T> : ObservableCollection<T>, IObserver<T>
    {

        #region IObserver<T> Members

        public void OnCompleted()
        {
         //   throw new NotImplementedException();
        }

        public void OnError(Exception exception)
        {
           // throw new NotImplementedException();
        }

        public void OnNext(T value)
        {
            this.Add(value);
        }

        #endregion
    }
}

File: ObservableSensor/ObservableClient/ObserverCollection.vb
Imports System.Collections.ObjectModel

Public Class ObserverCollection(Of T)
    Inherits ObservableCollection(Of T)
    Implements IObserver(Of T)


    Public Sub OnCompleted() Implements System.IObserver(Of T).OnCompleted

    End Sub

    Public Sub OnError(ByVal [error] As System.Exception) Implements System.IObserver(Of T).OnError

    End Sub

    Public Sub OnNext(ByVal value As T) Implements System.IObserver(Of T).OnNext
        Me.Add(value)
    End Sub
End Class
